src/dagmc.cpp

```diff
@@ -70,34 +160,68 @@ void load_dagmc_geometry()
     }
 
     if (model::DAG->is_implicit_complement(vol_handle)) {
-      // assuming implicit complement is void for now
+      // assuming implicit complement is always void
       c->material_.push_back(MATERIAL_VOID);
       continue;
     }
 
-    if (model::DAG->has_prop(vol_handle, "mat")){
-      std::string mat_value;
-      rval = model::DAG->prop_value(vol_handle, "mat", mat_value);
+    // check for temperature assignment
+    std::string temp_value;
+    if (model::DAG->has_prop(vol_handle, "temp")) {
+      rval = model::DAG->prop_value(vol_handle, "temp", temp_value);
       MB_CHK_ERR_CONT(rval);
-      to_lower(mat_value);
+      double temp = std::stod(temp_value);
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * temp));
+    } else {
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * settings::temperature_default));
+    }
 
-      if (mat_value == "void" || mat_value == "vacuum") {
-        c->material_.push_back(MATERIAL_VOID);
-      } else {
-        c->material_.push_back(std::stoi(mat_value));
-      }
+    // determine volume material assignment
+    std::string mat_value;
+    if (model::DAG->has_prop(vol_handle, "mat")) {
+      rval = model::DAG->prop_value(vol_handle, "mat", mat_value);
+      MB_CHK_ERR_CONT(rval);
     } else {
       std::stringstream err_msg;
       err_msg << "Volume " << c->id_ << " has no material assignment.";
       fatal_error(err_msg.str());
     }
+
+    std::string cmp_str = mat_value;
+    to_lower(cmp_str);
+    // material void checks
+    if (cmp_str.find("void") != std::string::npos   ||
+        cmp_str.find("vacuum") != std::string::npos ||
+        cmp_str.find("graveyard") != std::string::npos) {
+      c->material_.push_back(MATERIAL_VOID);
+    } else {
+      if (using_uwuw) {
+        // lookup material in uwuw if the were present
+        std::string uwuw_mat = DMD.volume_material_property_data_eh[vol_handle];
+        if (uwuw.material_library.count(uwuw_mat) != 0) {
+          int matnumber = uwuw.material_library[uwuw_mat].metadata["mat_number"].asInt();
```

Comment:
I would add a comment to remind the developer that from UWUW material numbers are arbitrary.

Reply:
Can do!

Reply:
Wellll I'm not sure about that. Right now I think we're relying on those numbers existing when we write the `materials.xml` file. Should we be using our own set of material ids and ignoring any existing ids?

Reply:
I think I wasn't specific enough, the id's are set, guaranteed to be unique, and monotonically increasing. They are numbered 1->N on the order they are loaded, the existing material number if it exists in the metadata section of the object is ignored.

Reply:
Ok that makes more sense. Adding that comment now.